File: 평강/app/project/src/main/java/com/eventpurchase/project/global/common/response/SuccessResponse.java
package com.eventpurchase.project.global.common.response;

import org.springframework.http.HttpStatus;

public record SuccessResponse<T> (
        int status,
        String message,
        T data
)
    {

        public static <T> SuccessResponse<T> of(HttpStatus status, String message, T data) {
            return new SuccessResponse<T>(status.value(), message, data);
        }
}

File: 평강/app/project/shared/src/main/kotlin/com/eventpurchase/project/shared/response/SuccessResponse.kt
package com.eventpurchase.project.shared.response

data class SuccessResponse<T>(
    val status: Int,
    val message: String,
    val data: T?
) {
    companion object {
        fun <T> of(status: Int, message: String, data: T?) = SuccessResponse(status, message, data)
    }
}
